server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -76,27 +76,35 @@ public static void main(String[] args)
                 font-src 'self' data: ${FONT.SOURCES} ;
                 script-src 'unsafe-eval' 'strict-dynamic' 'nonce-${REQUEST.SCRIPT.NONCE}' ;
                 base-uri 'self' ;
+                frame-src 'self' ${FRAME.SOURCES} ;
+            """;
+        // Add upgrade_insecure_requests substitution, frame-ancestors, and e12 version for enforce CSP
+        String enforceCsp = baseCsp + """
                 ${UPGRADE.INSECURE.REQUESTS}
```

Comment:
does this not need a ; at the end?

Reply:
The substitution adds the ;

This way, if running without HTTPS we don't get an extra semicolon

Reply:
@labkey-willm https://github.com/LabKey/platform/blob/develop/api/src/org/labkey/filters/ContentSecurityPolicyFilter.java#L300